### PR TITLE
chore: migrate rollup-plugin-typescript

### PR DESCRIPTION
**Pull request by NotWoods**
## Rollup Plugin Name: `@rollup/plugin-typescript`

This PR contains:

- [ ] bugfix
- [ ] feature
- [ ] refactor
- [ ] documentation
- [x] other

Are tests included?

- [x] yes (_bugfixes and features will not be merged without tests_)
- [ ] no

Breaking Changes?

- [x] yes (_breaking changes will not be merged unless absolutely necessary_)
- [ ] no

List any relevant issue numbers:

### Description

Starting work on migrating the TypeScript plugin 😄

### CODE REVIEW

**Comment by shellscape** [comment]
Huge. Thanks for getting this in. I need to wrap up the parent repo, and will do that hopefully today and archive it.

**Comment by tolu** [comment]
I'm still happy to help with moving `compilerOption` reads into the `startBuild` hook and fixing the `console.log/warning` (and anything really).

Just let me know how I can help! 🙏 😄

**Comment by shellscape** [comment]
@tolu Awesome. When the move is complete we'll ping you for sure.

**Comment by shellscape** [comment]
@NotWoods the repo is archived. we'll need to pull in the changes from that last PR and then I think we're good to review.

File: packages/typescript/src/index.js [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+  // Verify that we're targeting ES2015 modules.
+  const moduleType = opts.module.toUpperCase();
+  if (
+    moduleType !== 'ES2015' &&
+    moduleType !== 'ES6' &&
+    moduleType !== 'ESNEXT' &&
+    moduleType !== 'COMMONJS'
+  ) {
+    throw new Error(
+      `rollup-plugin-typescript: The module kind should be 'ES2015' or 'ESNext, found: '${opts.module}'`
```
Comment (pnevares) [inline packages/typescript/src/index.js]: Package should be renamed to `@rollup/plugin-typescript` throughout this file.

File: packages/typescript/test/test.js [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+});
+
+test('should throw on bad options', (t) => {
+  t.throws(
+    () =>
+      rollup({
+        input: 'does-not-matter.ts',
+        plugins: [typescript({ foo: 'bar' })]
+      }),
+    "rollup-plugin-typescript: Couldn't process compiler options"
```
Comment (pnevares) [inline packages/typescript/test/test.js]:
```suggestion
    "@rollup/plugin-typescript: Couldn't process compiler options"
```

File: packages/typescript/package.json [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    "tslib": "*",
+    "typescript": ">=2.1.0"
+  },
+  "dependencies": {
+    "resolve": "^1.10.0",
+    "@rollup/pluginutils": "^3.0.0"
+  },
+  "devDependencies": {
+    "buble": "^0.19.8",
+    "mocha": "^6.1.4",
```
Comment (pnevares) [inline packages/typescript/package.json]: Shouldn't need mocha if the ava switch is complete.

**Comment by NotWoods** [comment]
Whoops, I forgot to update the README. Thanks for catching that!

**Comment by shellscape** [comment]
@NotWoods this pnpm-lock conflict thing is a PITA. here's how to fix:

```console
git fetch upstream
git merge upstream/master
# will probably see a conflict here
rm pnpm-lock.json
pnpm install --force
git add . && git commit
# save commit
git push
```

That ought to do it. Once that's done we can merge.

**Comment by NotWoods** [comment]
Rebased and fixed conflicts

**Comment by shellscape** [comment]
thanks!

**Comment by RealAlphabet** [comment]
The "Why" section of the README makes no sense. Nor is there any additional information about this plugin in the babel plugin README.

**Comment by shellscape** [comment]
This PR is four years old. If you have a documentation issue, please open a new issue using the Documentation template. You may also find articles about contributing and interacting on GitHub to be useful.

**Pull request merged**